src/scanoss/agent/snippet_scan.c

```diff
@@ -0,0 +1,477 @@
+       extract_csv(detUrl, line, 9,lenLine,',');
+     
+      extract_csv(detMatch, line, 10,lenLine,',');
+      extract_csv(detLines, line, 12,lenLine,',');
+      extract_csv(detPath, line, 13,lenLine,',');
+       
+      
+      // License store
+      for (int i=1;i<5;i++){
+        char aux[100];
+        extract_csv(aux, detLicenses, i,100,';');
+        if (aux==NULL) break; 
+          else {    
+            int detLic= getLicenseId((unsigned char *) aux); /* ... from name, get the key of license that matches short_name at license_ref */
+            if(detLic!=0){    /** If the key is valid, insert the result on DB Table */
+              sprintf(auxSql,"INSERT INTO license_file(rf_fk, agent_fk, rf_timestamp, pfile_fk) VALUES(%d,%d, now(), %ld);",detLic,Agent_pk,key);
+              result = PQexec(db_conn,auxSql);
+            } else {/* No license found by its spdx id, lets check if it is a deprecated license */
+                struct node *dep = find(aux);
+                if(dep) {
+                int oldLic= getLicenseId((unsigned char *)dep->oldName);
+                if(oldLic!=0){ /* It is a deprecated identifier */
+                  sprintf(auxSql,"INSERT INTO license_file(rf_fk, agent_fk, rf_timestamp, pfile_fk) VALUES(%d,%d, now(), %ld);",oldLic,Agent_pk,key);
+                  result = PQexec(db_conn,auxSql);
+                } 
+                else 
+                  {
+                  /* It is a deprecated license but it is not included on DB */
+                  }
+            }
+          }
+        }
+      }
+      //File info store  
+      if(strcmp((char *)detectedUsage,"none") && (!(strcmp((char *)detectedUsage,"file"))||!(strcmp((char *)detectedUsage,"snippet")) )){  
+        char *auxSQL;
+        asprintf(&auxSQL,"INSERT INTO scanoss_fileinfo (pfile_fk, matchtype, lineranges, purl,filepath,url) VALUES(%ld, '%s', '%s', '%s','%s','%s');", key,detectedUsage,detLines,detPurls,detPath,detUrl);//,url,filePath);
+        result = PQexec(db_conn,auxSQL);
+        free(auxSQL);
+        if (PQntuples(result) == 0)
+        { 
+         PQclear(result);
+        }
+      } 
+      resCount++;
+     } else {break;}
+    }
+  }
+
+
+/*!
+ * \brief Scans a Temporary folder 
+ * \details Scans a Temporary folder with a rebuild project and place it results on a file results.csv
+ * \param folder path to temp folder
+ */
+int ScanFolder(char *folder)
+{
+
+  FILE *Fin;
+  char Cmd[MAXCMD];
+  memset(Cmd, '\0', MAXCMD);
+
+  unsigned char apiurl[400];
+  unsigned char key[110];
+ 
+  if(ApiUrl[0]!='\0')  {
+      sprintf((char *) apiurl,"--apiurl %s",ApiUrl);
+  }  
+  else  
+      memset(apiurl,0,sizeof(apiurl));
+
+  if(accToken[0]!='\0' && accToken[0]!=' ')  {
+    sprintf((char *)key,"--key %s",accToken);
+  }  
+  else  
+        memset(key,0,sizeof(key));
+  sprintf(Cmd, "scanoss-py scan  %s --format=csv -o %s/results.csv %s %s", folder, folder,apiurl,key); /* Create the command to run */
```

Comment:
Since fossology dependencies are installed in custom folder in `fo-install-pythondeps`, building the pythonpath will be required before calling the script. Please have a look at scancode's wrapper:
https://github.com/fossology/fossology/blob/f985e9b62d5d0be65143812805e2e5388ea01483/src/scancode/agent/scancode_wrapper.cc#L74-L81

Reply:
Applyed